### PR TITLE
[Fix] Add synonymes to some comet enzymes so PepXML reader recognizes them

### PR DESCRIPTION
**Pull request by jpfeuffer**
Also fixes a wrong annotation of unspecific cleavage for the crux enzymes.
Also makes the lookup for enzymes supported by OMSSA and Comet a bit more robust by making their IDs ints and introducing a specific "-1" for unsupported enzymes.
Removed hardcoded strings in pepXML reader.

### CODE REVIEW

**Comment by jpfeuffer** [comment]
Fixes #3191

File: src/openms/source/CHEMISTRY/DigestionEnzymeProtein.cpp [hunk 1]
```diff
@@ -46,10 +46,10 @@ namespace OpenMS
     c_term_gain_(""),
     psi_id_(""),
     xtandem_id_(""),
-    comet_id_(),
-    crux_id_(),
+    comet_id_(-1),
```
Comment (timosachsenberg) [inline src/openms/source/CHEMISTRY/DigestionEnzymeProtein.cpp]: not that important but in the future, we should transform these into =default constructors and define the default initialization value of the variables in the header.  Less error prone...

**Comment by jpfeuffer** [comment]
hmm will fix the test tomorrow.

File: share/OpenMS/CHEMISTRY/Enzymes.xml [hunk 2]
```diff
@@ -318,9 +328,14 @@
       <ITEM name="CTermGain" value="OH" type="string" />
       <ITEM name="PSIID" value="MS:1001956" type="string" />
       <ITEM name="XTandemID" value="[X]|[X]" type="string" />
+      <ITEM name="CruxID" value="no-enzyme" type="string" />
```
Comment (hroest) [inline share/OpenMS/CHEMISTRY/Enzymes.xml]: hmm, do you think this is what `no-enzyme ([X]|[X])` means? I thought that would be `([]|[])` if it cuts everywhere but we can ask

**Comment by jpfeuffer** [comment]
So the latest changes include a major rewrite of the CometAdapter parameters. Some had wrong types. I also added more documentation and some I think important parameters.
Also terminal modifications were written wrongly to the params file.
While searching for errors in the build log, I catched a warning in MSFraggerAdapter and found another small bug.
Adapted the tests accordingly.

File: src/topp/CometAdapter.cpp [hunk 3]
```diff
@@ -123,87 +123,97 @@ class TOPPCometAdapter :
...
     registerInputFile_("default_params_file", "<file>", "", "Default Comet params file. All parameters of this take precedence. A template file can be generated using comet.exe -p", false, false, ListUtils::create<String>("skipexists"));
     setValidFormats_("default_params_file", ListUtils::create<String>("txt"));
-    //registerIntOption_("threads", "<num>", 1, "number of threads", false, true);
 
     //Masses
-    registerDoubleOption_("precursor_mass_tolerance", "<tolerance>", 10.0, "Precursor monoisotopic mass tolerance (Comet parameter: peptide_mass_tolerance)", false, false);
-    registerStringOption_("precursor_error_units", "<choice>", "ppm", "peptide_mass_units 0=amu, 1=mmu, 2=ppm", false, false);
+    registerDoubleOption_("precursor_mass_tolerance", "<tolerance>", 10.0, "Precursor monoisotopic mass tolerance (Comet parameter: peptide_mass_tolerance).  See also precursor_error_units to set the unit.", false, false);
+    registerStringOption_("precursor_error_units", "<choice>", "ppm", "Unit of precursor monoisotopic mass tolerance for parameter precursor_mass_tolerance (Comet parameter: peptide_mass_units)", false, false);
```
Comment (timosachsenberg) [inline src/topp/CometAdapter.cpp]: is "..._units" correct? would _unit make more sense

**Comment by jpfeuffer** [comment]
Yes will change. I Have another commit coming.

**Comment by jpfeuffer** [comment]
If that runs through. Ready to merge.

**Comment by timosachsenberg** [comment]
+1 if passes

**Comment by jpfeuffer** [comment]
Windows unrelated and macOS the unmatching boost problems.

**Comment by Matthias313** [comment]
To test if the the CometAdapter now runs through, I will need the developer version (nightly version) of OMS?

**Comment by jpfeuffer** [comment]
Yes exactly. Do you build it yourself or do you need an installer? Do you use the KNIME plugin?

**Comment by Matthias313** [comment]
@lars20070 
Lars updated OpenMS on our Server (2.3). 
Is there the option to only update some tools (e.g. the CometAdapter)???
Currently, I am using TOPPAS, since I am starting the runs via a batch skript which is not KNIME compatible and I don´t have a lot of experience using KNIME.

**Comment by Matthias313** [comment]
Alright, I will wait for Lars.
Thanks for the Info and the fast fix of the bug @jpfeuffer !

**Pull request merged**